spec/rubocop/cop/rspec/hooks_before_examples_spec.rb

```diff
@@ -227,4 +227,65 @@
       RUBY
     end
   end
+
+  context 'when Ruby 3.4', :ruby34 do
+    it 'flags `around` after `it`' do
+      expect_offense(<<~RUBY)
+        RSpec.describe User do
+          it { is_expected.to be_after_around_hook }
+          around { it }
```

Comment:
I wonder if this would work anyway without making a special case for an it-block? Looks like a regular no-args block call with an ‘it’ call inside. No surprise, this may or may not be an it-block, hard to tell.

Side note, how does Ruby knows ‘describe { it { foo } }’ is not an it-block?

Reply:
By what I propose here to keep the spec, and attempt to remove special handling in the cop. Chances are it will work anyway.

Reply:
Yup, that's what I did: https://github.com/rubocop/rubocop-rspec/pull/2077#discussion_r2039275425

> Side note, how does Ruby knows ‘describe { it { foo } }’ is not an it-block?

`it` with block had special considerations from ruby itself, `it` only points to the block argument when `it` doesn't take a block itself. https://bugs.ruby-lang.org/issues/18980
> The following cases have been discussed:
>  *  it method, most famously in RSpec: You almost always pass a positional and/or block argument to RSpec's it, so the conflict is avoided with my proposal. You virtually never use a completely naked it